lib/bitcoin/lock/LockMonitor.ts

```diff
@@ -0,0 +1,304 @@
+import BitcoinClient from '../BitcoinClient';
+import BitcoinError from '../BitcoinError';
+import BitcoinLockTransactionModel from '../models/BitcoinLockTransactionModel';
+import ErrorCode from '../ErrorCode';
+import LockIdentifier from '../models/LockIdentifierModel';
+import LockIdentifierSerializer from './LockIdentifierSerializer';
+import LockResolver from './LockResolver';
+import LockTransactionModel from './../models/LockTransactionModel';
+import LockTransactionType from './../enums/LockTransactionType';
+import MongoDbLockTransactionStore from './MongoDbLockTransactionStore';
+import ValueTimeLockModel from './../../common/models/ValueTimeLockModel';
+
+/**
+ * Structure (internal to this class) to track the information about lock.
+ */
+interface LockInformation {
+  currentValueTimeLock: ValueTimeLockModel | undefined;
+  latestSavedLockInfo: LockTransactionModel | undefined;
+
+  // Need to add a 'state'
+}
+
+/**
+ * Encapsulates functionality to monitor and create/remove amount locks on bitcoin.
+ */
+export default class LockMonitor {
+
+  private periodicPollTimeoutId: number | undefined;
+
+  private currentValueTimeLock: LockInformation | undefined;
+
+  private lockResolver: LockResolver;
+
+  constructor (
+    private bitcoinClient: BitcoinClient,
+    private lockTransactionStore: MongoDbLockTransactionStore,
+    private pollPeriodInSeconds: number,
+    private desiredLockAmountInSatoshis: number,
+    private lockPeriodInBlocks: number,
+    private firstLockFeeAmountInSatoshis: number) {
+
+    this.lockResolver = new LockResolver(this.bitcoinClient);
+  }
+
+  /**
+   * Initializes this object by performing the periodic poll tasks.
+   */
+  public async initialize (): Promise<void> {
+    await this.periodicPoll();
+  }
+
+  private async periodicPoll (intervalInSeconds: number = this.pollPeriodInSeconds): Promise<void> {

```

Comment:
Why not just use this.pollPeriodInSeconds all the time? Feels weird that we can override it.

Reply:
Updated to use the member variable.